### ISSUE TITLE
Add "Pause" option for pausing in the middle of a layer (bottom slider)

### ISSUE THREAD

**Issue by misterjmw** [SoftFever/OrcaSlicer#6334]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

I am frequently annoyed when I have to add pauses mid-layer to a print, in-order to change filament colors because the slicer doesn't have the ability to do so.

### Which printers will be beneficial to this feature?

Klipper, Marlin, Others

### Describe the solution you'd like

I am still a lowly plebe with a single material printer. However, it's extremely useful to print in multiple colors. Example; I'm cutting text into a model and I want the background of the text to be contrasting and stand out once printed on top of. 

However, if I pause the layer, change filaments and then do the reverse for the next layer, the entire layer must be printed in that contrasting color. 

However **if** I could add a pause/filament change on the horizontal, mid-layer bar, I could trigger a pause before the section printed, then pause again after to swap back, thus only the feature I want is in the contrasting color.

While I think this pause is useful, I could alternatively see this as functionally being integrated into the 'paint' feature, so that you could paint a feature a color, and the slicer would group/organize pauses for it knowing that the printer profile does not have an MMU/AMS. 

### Describe alternatives you've considered

Alternatives considered; 
The only option for this (that I'm aware of) is to edit the gcode by hand, adding M25/filament change. However this is tedious because you have to know exactly where to pause.

### Additional context

This would help the majority of users who do not have MMU/AMS units.

**Comment by iamapig120** [comment]
![image](https://github.com/user-attachments/assets/ee51e50c-8ed5-4ec7-9a10-d8e24602f6ff)
You could use this option to enable change fliament by yourself.

**Comment by misterjmw** [comment]
@iamapig120 I do not follow how that achieves the goal. Does that allow the UI to add a 'filament change' anywhere? much like you can add a pause via right click, on the per-layer slider (on the right)?

**Comment by chatasos** [comment]
Hi,

This is something i would also be interested in.

**Comment by dakotawinslow** [comment]
I would also love to see this feature!

**Comment by simpat1zq** [comment]
Just went searching to see how to do this and found this post. So I guess it answers my question that it's not possible. I'd love to see this as well.

**Comment by bolens** [comment]
This feature would be really useful

**Comment by Thomas49600** [comment]
Agreed, waiting around until my printer starts to print a purge tower where I then must manually pause and restart the print is a bit of a pain... especially if layers take a long time to print and I need to babysit the machine whilst I wait so that I don't miss the 10second pause window.

**Comment by jnse78** [comment]
This would be a great function 👍

**Comment by bigfoott** [comment]
I'd also love this!!

**Comment by designnoevil** [comment]
yes adding a paurse mid layer, not just at new layer, would be very useful

**Comment by aquapura001** [comment]
I'd love this feature too please. I need to be able to pause mid layer to remove threading when doing multi color.

**Comment by Tolvelski** [comment]
I would also like this feature, but my reasoning is for calibration purposes. 

For example, I would like to be able to have a group of small, single layer squares arranged close together, and after each square, pull the hotend aside and present the bed. This would allow me to see my manual Z-Offset adjustment results more easily on my Neptune 4 Max, which has a large hotend that obstructs my view of my adjustments until the head has moved out of the way. This means I have to either manually pause the printer, or keep running the same print over and over. Now the print itself doesn't take too long, but with KAMP, Homing, Heatsoaking, etc, every new print adds up. 

Being able to have 9 squares, close together, print one, pause, present, adjust Z-Offset, and then resume, would be nice.

**Comment by Ineedmoney420** [comment]
I know tool changers I becoming more popular but I see myself wanting this on multiple occasion having only a single tool at the moment..

**Comment by XBH3Bandit** [comment]
I would also be very interested in this feature for 5 colors on a single layer while using a single AMS.

I have a print for a lightbox which uses 5 colors per layer for the first 4 layers. I can obviously just keep swapping out the 2 colors that are furthest apart in the color changing sequence but it would be so much better to be able to just add a pause at each point within the layer when I need to swap out the colors.

**Comment by RhochR** [comment]
+1

**Comment by vargabp** [comment]
I would love to see this feature too.

**Comment by iRon185** [comment]
Glad others have been asking for this feature. I love the Pause feature on the vertical bar but when I realized it doesn't exist for the horizontal bar, my search brought me to this thread.

**Comment by Jorazon** [comment]
I would also like this feature.

**Comment by christopher-lambe** [comment]
This would be great for embedding magnets too. The nozzle likes to ooze when paused.
Pause somewhere in the infill after the outside walls are printed.
Insert magnets.
Nozzle is oozing slightly - quickly wipe it with brass brush
Resume. The infill on that layer might be little imperfect, but the outside walls will not suffer because of nozzle pressure loss from the pause oozing

**Comment by Incendos** [comment]
Yeah, just searched for this feature for the same purpose: wanted to pause mid-layer to drop some magnets in it.

**Comment by christopher-lambe** [comment]
Workaround for mid layer pause for magnet insertions

Yes we all know how - insert a pause (right click on + in vertical slider) at the start of the layer that covers the magnet (the layer where the bridge is printed - the hole is shown as closed) and click slice again. But this is not good enough. The nozzle oozes and loses pressure the outside walls may suffer quality issues.

THE WORKAROUND

Set Timelapse to Smooth: This moves the head out of the way (so it not all over the place in the timelapse) and also importantly makes a prime tower. The nozzle is primed at the start of each layer. So you can insert the pauses as needed and any oozing is taken care of when the nozzle is primed on the tower. Just be quick when inserting the magnets so the print does not cool too much and you cause a hull line.

**Comment by christopher-lambe** [comment]
> Yeah, just searched for this feature for the same purpose: wanted to pause mid-layer to drop some magnets in it.

Workaround above - sorry forgot to quote you for my reply

**Comment by Incendos** [comment]
> > Yeah, just searched for this feature for the same purpose: wanted to pause mid-layer to drop some magnets in it.
> 
> Workaround above - sorry forgot to quote you for my reply

It is an acceptable workaround, but when i have a 250mm tall model printing and i want to drop magnets near the top, i don't really want to waste ~30g for a prime tower. I rather let the infill take the loss.

**Comment by christopher-lambe** [comment]
> It is an acceptable workaround, but when i have a 250mm tall model printing and i want to drop magnets near the top, i don't really want to waste ~30g for a prime tower. I rather let the infill take the loss.

Then wait and hope like I am or edit the G-Code in text editor and add the pause where you need it. Move sliders till the head is out of the way and note the line number in the G-Code window in OrcaSlicer paste the G-Code for the pause at this line number.

I have not done this yet and cannot tell you about dangers or caveats of doing this. Hope your print works out dude.

**Comment by charlesmx48** [comment]
I would also like a feature like this.

Feature Request: Add a "Pause" slider to the bottom horizontal printhead line, identical in functionality to the current right vertical layer slider. Users should be able to insert, delete, and visualize pauses not just by layer, but within a layer (at the beginning, middle, or end).

Current Pain Point: Right now, inserting a pause gives no visual feedback on whether it occurs at the start or end of the layer. Having to dig into the G-code window to figure it out is highly inefficient and cumbersome.

**Comment by kingofpotatopeople-ui** [comment]
This was discussed and a solution posted on makerworld:
https://makerworld.com/en/models/978625-manual-ams#profileId-951912

**Comment by misterjmw** [comment]
@kingofpotatopeople-ui 
I do not see how that thread applies to this topic. This is a request for an **interface** change.

If we just needed a hack to add a pause to a section, or autopause during a filament change that's simple Gcode.

**Comment by misterjmw** [comment]
Keeping this alive. Still waiting for a mid-layer pause feature.
